ray_on_golem/server/run.py

```diff
@@ -128,29 +157,60 @@ async def ray_service_ctx(app: web.Application) -> None:
     await ray_service.shutdown()
 
 
+@asynccontextmanager
+async def golem_network_stats_service(registry_stats: bool) -> GolemNetworkStatsService:
+    golem_network_stats_service: GolemNetworkStatsService = GolemNetworkStatsService(registry_stats)
+    yagna_service = YagnaService(
+        yagna_path=YAGNA_PATH,
+    )
+
+    await yagna_service.init()
+    await golem_network_stats_service.init(yagna_appkey=yagna_service.yagna_appkey)
+
+    yield golem_network_stats_service
+
+    await golem_network_stats_service.shutdown()
+    await yagna_service.shutdown()
+
+
+async def stats_main(args: argparse.Namespace):
+    with args.CLUSTER_CONFIG_FILE as file:
+        config = yaml.safe_load(file.read())
+    provider_config = config["provider"]
+    async with golem_network_stats_service(
+        provider_config["enable_registry_stats"]
+    ) as stats_service:
+        await stats_service.run(provider_config["parameters"], args.run_time)
+
+
 def main():
     prepare_tmp_dir()
     args = parse_sys_args()
 
-    logging.config.dictConfig(LOGGING_CONFIG)
+    if args.command == "webserver":
+        logging.config.dictConfig(LOGGING_CONFIG)
```

Comment:
I'd move these conditional pieces to separate `webserver` and `stats` functions for clarity's sake